### ISSUE TITLE
FK relationship type guessing

### ISSUE THREAD

**Issue by nsajko** [pgmodeler/pgmodeler#1764]
There seem to be some inconsistencies when guessing foreign key relationship types, as far as I understand.

### The example from the documentation

Consider the example from the PGModeler documentation: https://www.pgmodeler.io/support/docs/310-relationships?v=1.0.0

![f34c330d30d26420d47677671db4f655](https://user-images.githubusercontent.com/4944410/227703308-7fdee40a-dced-4efa-8461-502ddcda81f3.png)

For example, the difference between `table_b` and `table_e` is that `table_e` has an additional *unique* constraint. However, the constraint isn't on the foreign key, it is rather on the primary key. But, AFAIK, the primary key constraint is the intersection of two constraints: non-null and unique. So the two tables are really equivalent! But why are their guessed FK relationship types  different, then, is this not a contradiction? I think you meant to check the *foreign key*, specifically, for uniqueness?

The situation is the same with tables `table_c` and `table_d`.

### Another example

DBM:
```
<?xml version="1.0" encoding="UTF-8"?>
<!--
CAUTION: Do not modify this file unless you know what you are doing.
 Unexpected results may occur if the code is changed deliberately.
-->
<dbmodel pgmodeler-ver="1.0.2" use-changelog="false" last-position="0,0" last-zoom="1" max-obj-count="4"
	 default-schema="public" default-owner="postgres"
	 layers="Default layer"
	 active-layers="0"
	 layer-name-colors="#000000"
	 layer-rect-colors="#b4b4b4"
	 show-layer-names="false" show-layer-rects="false">
<role name="pg_database_owner"
 inherit="true"
 password="********"
 sql-disabled="true">
</role>

<role name="nsajko"
 superuser="true"
 createdb="true"
 createrole="true"
 inherit="true"
 login="true"
 password="********">
</role>

<database name="studadmin" encoding="UTF8" lc-collate="C.UTF-8" lc-ctype="C.UTF-8" is-template="false" allow-conns="true">
	<role name="nsajko"/>
	<tablespace name="pg_default"/>
</database>

<schema name="public" layers="0" rect-visible="true" fill-color="#e1e1e1" sql-disabled="true">
</schema>

<table name="mjesto" layers="0" collapse-mode="2" max-obj-count="1" z-value="0">
	<schema name="public"/>
	<role name="nsajko"/>
	<position x="20" y="80"/>
	<column name="pbr" not-null="true">
		<type name="integer" length="0"/>
	</column>
	<constraint name="mjesto_pkey" type="pk-constr" table="public.mjesto">
		<columns names="pbr" ref-type="src-columns"/>
	</constraint>
</table>

<table name="student" layers="0" collapse-mode="2" max-obj-count="3" z-value="0">
	<schema name="public"/>
	<role name="nsajko"/>
	<position x="420" y="40"/>
	<column name="jmbag" not-null="true">
		<type name="character varying" length="10"/>
	</column>
	<column name="oib">
		<type name="character varying" length="11"/>
	</column>
	<column name="pbrstanstudent">
		<type name="integer" length="0"/>
	</column>
	<constraint name="student_pkey" type="pk-constr" table="public.student">
		<columns names="jmbag" ref-type="src-columns"/>
	</constraint>
	<constraint name="student_oib_key" type="uq-constr" table="public.student">
		<columns names="oib" ref-type="src-columns"/>
	</constraint>
</table>

<constraint name="student_pbrstanstudent_fkey" type="fk-constr" comparison-type="MATCH SIMPLE"
	 upd-action="NO ACTION" del-action="NO ACTION" ref-table="public.mjesto" table="public.student">
	<columns names="pbrstanstudent" ref-type="src-columns"/>
	<columns names="pbr" ref-type="dst-columns"/>
</constraint>

<relationship name="rel_student_mjesto1" type="relfk" layers="0"
	 custom-color="#3c3f90"
	 src-table="public.student"
	 dst-table="public.mjesto" reference-fk="student_pbrstanstudent_fkey"
	 src-required="false" dst-required="false"/>

<permission>
	<object name="public" type="schema"/>
	<roles names="pg_database_owner"/>
	<privileges create="true" usage="true"/>
</permission>
<permission>
	<object name="public" type="schema"/>
	<privileges usage="true"/>
</permission>
</dbmodel>
```

Screenshot:

![with_unique](https://user-images.githubusercontent.com/4944410/227704146-63e33fb2-e7b1-44ab-953c-f125d0791b5b.png)

If the unique constraint is removed, the guessed relationship type changes, even though the constraint was on a column that doesn't have *any* other constraints:

![without_unique](https://user-images.githubusercontent.com/4944410/227704236-f70f82d9-23c8-4101-b598-1eea5bf9ccfe.png)

**Comment by rkhaotix** [comment]
Hi @nsajko 

Thanks for reporting.

> I think you meant to check the foreign key, specifically, for uniqueness?

That's the case. Maybe we have a bug here. I'll investigate and se what's wrong.

**Comment by nsajko** [comment]
Indeed it seems that the bug is fixed on the 1.0.3 branch. Thanks!
